### ISSUE TITLE
Problems with DST change and cylc restart

### ISSUE THREAD

**Issue by dpmatthews** [cylc/cylc-flow#3462]
If you start a suite in one timezone (a fixed offset from UTC) it will continue running using that timezone regardless of any timezone change due to daylight savings (we should make sure this is clear in the documentation). Therefore, if you stop a suite and restart it you should expect it to continue using the original timezone. However, this is not what happens at present (problem showed up with a suite which got migrated between servers).

I used this suite for testing (with 7.8.4):
```ini
[cylc]
    UTC mode = False # No problem is this is set to True
[scheduling]
    initial cycle point = now
    [[special tasks]]
        clock-trigger = daily(PT0M)
    [[dependencies]]
        [[[T23]]]
            graph = daily
[runtime]
    [[daily]]
        script = true
```

Example output illustrating the problem below.
Note the change in cyclepoint for future cycles after the restart.
Also note that, after the restart, I don't appear to be able to interact with the existing task without using wildcards (the same happens in the GUI).

```
$ export TZ=UTC
$ cylc run test.tz --host=localhost
<snip>
$ cylc dump test.tz
last_updated=1575634329.66
namespace definition order=['daily', 'root']
newest cycle point string=20191206T2300Z
newest runahead cycle point string=None
oldest cycle point string=20191206T2300Z
reloading=False
run_mode=live
state totals={'waiting': 1}
states=['waiting']
status_string=running
suite_urls={'suite': '', 'root': '', 'daily': ''}
time zone info={'hours': 0, 'string_basic': 'Z', 'string_extended': 'Z', 'minutes': 0}
daily, 20191206T2300Z, waiting, unspawned
$ cylc show test.tz 20191206T2300Z/daily
title: (not given)
description: (not given)
URL: (not given)

prerequisites (- => not satisfied):
  (None)

outputs (- => not completed):
  - daily.20191206T2300Z expired
  - daily.20191206T2300Z submitted
  - daily.20191206T2300Z submit-failed
  - daily.20191206T2300Z started
  - daily.20191206T2300Z succeeded
  - daily.20191206T2300Z failed

other:
  o  Clock trigger time reached ... False
  o  Triggers at ... 2019-12-06T23:00:00Z
$ cylc stop test.tz
$ export TZ=BST-1
$ cylc restart test.tz --host=localhost
<snip>
$ cylc dump test.tz
last_updated=1575634369.05
namespace definition order=['daily', 'root']
newest cycle point string=20191206T2300Z
newest runahead cycle point string=None
oldest cycle point string=20191206T2300Z
reloading=False
run_mode=live
state totals={'waiting': 1}
states=['waiting']
status_string=running
suite_urls={'suite': '', 'root': '', 'daily': ''}
time zone info={'hours': 1, 'string_basic': '+01', 'string_extended': '+01:00', 'minutes': 0}
daily, 20191206T2300Z, waiting, unspawned
$ cylc show test.tz 20191206T2300Z/daily  # Why doesn't this work?
No matching tasks found: 20191206T2300Z/daily
$ cylc show test.tz 20191206T2300*/daily
title: (not given)
description: (not given)
URL: (not given)

prerequisites (- => not satisfied):
  (None)

outputs (- => not completed):
  - daily.20191206T2300Z expired
  - daily.20191206T2300Z submitted
  - daily.20191206T2300Z submit-failed
  - daily.20191206T2300Z started
  - daily.20191206T2300Z succeeded
  - daily.20191206T2300Z failed

other:
  o  Clock trigger time reached ... False
  o  Triggers at ... 2019-12-07T00:00:00+01:00
$ cylc trigger test.tz 20191206T2300*/daily
$ cylc dump test.tz
last_updated=1575634467.15
namespace definition order=['daily', 'root']
newest cycle point string=20191207T2300+01
newest runahead cycle point string=None
oldest cycle point string=20191207T2300+01
reloading=False
run_mode=live
state totals={'waiting': 1}
states=['waiting']
status_string=running
suite_urls={'suite': '', 'root': '', 'daily': ''}
time zone info={'hours': 1, 'string_basic': '+01', 'string_extended': '+01:00', 'minutes': 0}
daily, 20191207T2300+01, waiting, unspawned
```

**Comment by dpmatthews** [comment]
See also #2324

**Comment by MetRonnie** [comment]
Is the problem fixed by setting
```ini
[cylc]
    cycle point time zone = Z
```
?

**Comment by MetRonnie** [comment]
I have found that if you set e.g. `time_zone_hours_minutes = (0, 0)` in the last line here
https://github.com/cylc/cylc-flow/blob/8a9043187eb7948674de86f53aa039908a92ef0c/cylc/flow/cycling/iso8601.py#L800-L812

it will fix the problem and correctly trigger at T23Z instead of T23+01 (which is T22Z).

However, the only reference to `time_zone_hours_minutes` after this is
https://github.com/cylc/cylc-flow/blob/8a9043187eb7948674de86f53aa039908a92ef0c/cylc/flow/cycling/iso8601.py#L815

then
https://github.com/cylc/cylc-flow/blob/8a9043187eb7948674de86f53aa039908a92ef0c/cylc/flow/cycling/iso8601.py#L831-L843

but there is no reference to `SuiteSpecifics` oustide of this file.

So how on Earth does this end up affecting the time zone of task triggering?

**Comment by MetRonnie** [comment]
@dpmatthews Does the fix for this need to be retrofitted to 7.8.x?

**Comment by dpmatthews** [comment]
> @dpmatthews Does the fix for this need to be retrofitted to 7.8.x?

Not a priority but we can leave this issue open as a reminder for the moment.
